packages/graph-explorer/src/connector/openCypher/queries/fetchSchema.ts

```diff
@@ -140,30 +149,47 @@ const fetchEdgesAttributes = async (
     return edges;
   }
 
-  await Promise.all(labels.map(async labelResult => {
-    const edgesTemplate = edgesSchemaTemplate({
-      type: labelResult
-    });
-
-    const response = await openCypherFetch<RawEdgesSchemaResponse>(edgesTemplate);
-
-    const edge = response.results[0].object as OCEdge;
-    const label = edge["~entityType"] as string;
-    const properties = edge["~properties"];
-    edges.push({
-      type: label,
-      displayLabel: sanitizeText(label),
-      total: countsByLabel[label],
-      attributes: Object.entries(properties || {}).map(([name, prop]) => {
-        const value = prop;
-        return {
-          name,
-          displayLabel: sanitizeText(name),
-          dataType: typeof value === "string" ? "String" : "Number",
-        };
-      }),
-    });
-  }));
+  await Promise.all(
+    labels.map(async labelResult => {
+      const edgesTemplate = edgesSchemaTemplate({
+        type: labelResult,
+      });
+
+      const response = await openCypherFetch<RawEdgesSchemaResponse>(
+        edgesTemplate
+      );
+
+      if (!response.results || response.results.length === 0) {
+        console.warn("Skipping label because there are no edges:", labelResult);
+        return;
+      }
+
+      if (!response.results[0].object) {
+        console.warn(
+          "Skipping label because the edge has no definition:",
+          labelResult
+        );
+        return;
+      }
+
+      const edge = response.results[0].object as OCEdge;
+      const label = edge["~entityType"] as string;
+      const properties = edge["~properties"];
+      edges.push({
+        type: label,
+        displayLabel: sanitizeText(label),
+        total: countsByLabel[label],
+        attributes: Object.entries(properties || {}).map(([name, prop]) => {
+          const value = prop;
+          return {
+            name,
+            displayLabel: sanitizeText(name),
```

Comment:
Is it sufficient to avoid possible code injections into UI?

Reply:
This by itself is not sufficient. The `sanitizeText()` method is modifying the name value to make it consistent in the UI (capitalization, remove extra spaces, etc).

React should prevent html injection. But I am not familiar enough with the codebase to know if JavaScript injection is possible. I don't see `eval()` used any where in the codebase, so that is good.

Reply:
I ran through some test cases that @alexey-temnikov provided and I did not see any negative impact in the UI. From what I can see, we are in good shape.